test/cpp/end2end/grpc_authz_end2end_test.cc

```diff
@@ -615,32 +606,33 @@ TEST_F(GrpcAuthzEnd2EndTest, StaticInitWithAuditLoggingOnAllow) {
   grpc::Status status;
 
   ClientContext context1;
+  // Matches the allow rule.
   context1.AddMetadata("key-foo", "foo");
-  ::testing::internal::CaptureStdout();
   status = SendRpc(channel, &context1, &resp);
-  CompareAuditLogEntry(
-      ::testing::internal::GetCapturedStdout(),
-      "{\"authorized\":true,\"matched_rule\":\"allow_foo\",\"policy_name\":"
-      "\"authz\","
-      "\"principal\":\"spiffe://foo.com/bar/baz\",\"rpc_method\":\"/"
-      "grpc.testing.EchoTestService/Echo\"}");
   EXPECT_TRUE(status.ok());
-  EXPECT_EQ(log_count_, 1);
+  EXPECT_EQ(audit_logs_.size(), 1);
+  EXPECT_THAT(
+      audit_logs_,
+      ::testing::ElementsAre(
+          "{\"authorized\":true,\"matched_rule\":\"allow_foo\",\"policy_"
+          "name\":\"authz\",\"principal\":\"spiffe://foo.com/bar/"
+          "baz\",\"rpc_method\":\"/grpc.testing.EchoTestService/Echo\"}"));
 
   ClientContext context2;
+  // Does not match the allow rule.
```

Comment:
```suggestion
  // Does not match the allow rule. Does not emit an audit log.
```

Reply:
Please add to each spot that is designed not to emit an audit log.
I think the readability is significantly better with the slight comment addition.

My main nit here is that you have to read the whole test case to get the context for the checks - for example, here I only know that this is _not_ supposed to emit a log because the previous RPC emitted one, so the count was already `1` by the time the check on line 627 is done.
Adding a few comments will balance that enough for a test

Reply:
Added the comment.